overlord/ifacestate/ifacestate.go

```diff
@@ -160,6 +160,16 @@ func connect(st *state.State, plugSnap, plugName, slotSnap, slotName string, fla
 	// slot's attributes in the *-slot-* hook).
 	// 'snapctl get' can read both slot's and plug's attributes.
 
+	// check if the connection already exists
+	conns, err := getConns(st)
+	if err != nil {
+		return nil, err
+	}
+	connRef := &interfaces.ConnRef{PlugRef: interfaces.PlugRef{Snap: plugSnap, Name: plugName}, SlotRef: interfaces.SlotRef{Snap: slotSnap, Name: slotName}}
+	if _, ok := conns[connRef.ID()]; ok {
+		return state.NewTaskSet(), nil
```

Comment:
given that this is not a *Many operation maybe returning a ifacestate.ErrAlreadyConnected would be preferable?